crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -602,6 +608,42 @@ async fn health_checks(State(connection): State<ConnectionConfig>) -> StatusCode
     }
 }
 
+#[derive(serde::Deserialize)]
+struct HealthParam {
+    max_checkpoint_ts_lag: Option<u64>,
+}
+
+/// Endpoint for querying the health of the service.
+/// It returns 500 for any internal error, including not connecting to the DB,
+/// and 503 if the checkpoint timestamp is too far behind the current timestamp as per the
```

Comment:
Is 504 ("Gateway Timeout") a more appropriate error code? I think we also get 503 from arbitrary panics so it might be good to differentiate from that and the rationale for picking 504 is that the GraphQL service is acting as a gateway to the DB, which is taking longer than the provided timeout parameter to respond.

Reply:
Sure, let's use 504 instead.